Relicense from LGPL-3 to BSD-3

lbdreyer opened the pull request:
All the code in this repo comes from contributors who have either signed the new CLA or are employees of the Met Office. The new CLA enables us to relicense to BSD-3.

This PR add ta BSD-3 LICENSE file (to replace the old COPYING* files)

I've created a draft PR as I just need to check the year in the LICENSE file is correct

bjlittle reviewed (changes requested):
@lbdreyer This is awesome! Thanks :+1: 

A couple of points:
* I think that we should additionally remove the entire mention of `© British Crown Copyright 2014-2022, Met Office` from the `README.md`. What do you think?
* You'll need to update the trove classifier in the `setup.cfg` to be `License :: OSI Approved :: BSD License` instead
* Also in `setup.cfg` you'll need to do `license_files = LICENSE` 
* Rebasing of #85 should fix your other issues

lbdreyer commented:
Thanks for reviewing @bjlittle !

I have addressed your review comments. I also ran a grep search for all other mentions of "(L)GPL" and "COPYING" and found that the headers needed updating as well. I believe that should have caught everything now!

bjlittle commented:
@lbdreyer LGTM :+1: 

As an additional though, we should also update the associated conda-forge [recipe](https://github.com/conda-forge/tephi-feedstock/blob/main/recipe/meta.yaml#L34-L38) :+1:
